<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/><meta name="exporter-version" content="Evernote Mac 6.0.7 (451372)"/><meta name="keywords" content="datatime"/><meta name="author" content="ashley17061@gmail.com"/><meta name="created" content="2015-03-11 17:11:34 +0000"/><meta name="source" content="desktop.mac"/><meta name="updated" content="2015-03-22 14:31:55 +0000"/><title>20150315_DataTime</title></head><body>
<div>Summaries for JGI initiation meeting:</div>
<div><br/></div>
<div>1.  Amplicon sequencing:  2X150 bp paired end reads over 250 bp region; 4,747,374 total reads</div>
<div>Jackson’s analysis on even_144K sequences/sample for five samples (3 replications of Cen13 culture-independent (CI) sample extracted with MO BIO, 1 replication of Cen13 CI sample extracted with another kit (Fast Prep?), and 1 replicate of Cen13 thermophile scraped pool of 60 isolates)</div>
<div>     Cen13 - scraped - 2494 OTUs from raw 2277,209 sequences</div>
<div>     Cen13 - CI (average across MO BIO) - 11,139 from 316,076 raw sequences</div>
<div>     Jackson checked, the second DNA extraction kit did not yield mutually exclusive OTUs...</div>
<div>     Note:  make a table of this - how many sequences we start out with for each sample, and how many are lost at each step in QC and analysis (e.g., after pandaseq merging, after chimera checking, after alignment, etc</div>
<div>     Sang-Hoon also performed a rarefaction to 77,000 seq/sample (with singletons included) and performed preliminary analysis (ordination) of the data, which shows clear distinction between active vents and recovered/reference soils!</div>
<div><br/></div>
<div>2.  Untargeted (shotgun) sequencing on Cen13 scraped pool and Cen13 CI DNA (combined across all 3 replicate extractions)</div>
<div>     Josh has been working through diginorm and genome assembly and annotation using khmer protocol.  unique khmer/sequencing depth asymptotes ~78,000,000  target sequencing depth; Josh estimates that we will have draft genome assemblies of some of the Centralia thermophiles</div>
<div>     -total reads = 421,376,543</div>
<div><br/></div>
<div>Other notes:</div>
<div>We will table the mock community analysis for now, but revisit later.</div>
<div><br/></div>
<div>Nice work everyone!</div>
<div><br/></div>
<div>Upcoming data deadlines:</div>
<div>1.  <b>By Mar 18 at the very latest</b> (sooner is better) - figures and summary statements about preliminary analyses.  Jackson is looking at Centralia thermophile diversity using 16S genes, Josh is looking at kmer/assembly/annotation (possible search for rpf and gyrA), Sang-Hoon is looking at changes along the Centralia gradient</div>
<div>2.  <b>By April 1</b> - figure for NSF grant submission</div>
<div><br/></div>
<div><br/></div>
<div><br/></div>
<div><br/></div>
</body></html>